PROTO_tests/tests/message_test.py

```diff
@@ -0,0 +1,234 @@
+            json.dumps({"key": "fields", "objType": "PDARRAY", "dtype": "uint64", "val": f.name})
+        ])
+        assert args == expected
+
+    @pytest.mark.parametrize("dtype", PDA_TYPES)
+    def test_pda_arg(self, dtype):
+        pda1 = ak.arange(3, dtype=dtype)
+        pda2 = ak.arange(4, dtype=dtype)
+        size, args = _json_args_to_str({"pda1": pda1, "pda2": pda2})
+        expected = json.dumps([
+            json.dumps({"key": "pda1", "objType": "PDARRAY", "dtype": str(pda1.dtype), "val": pda1.name}),
+            json.dumps({"key": "pda2", "objType": "PDARRAY", "dtype": str(pda2.dtype), "val": pda2.name})
+        ])
+        assert args == expected
+
+        size, args = _json_args_to_str({"pda_list": [pda1, pda2]})
+        expected = json.dumps([
+            json.dumps({"key": "pda_list", "objType": "LIST", "dtype": ak.pdarray.objType, "val": json.dumps([pda1.name, pda2.name])}),
+        ])
+        assert args == expected
+
+    def test_segstr_arg(self):
+        str1 = ak.array(["abc", "def"])
+        str2 = ak.array(["Test", "Test2"])
+        size, args = _json_args_to_str({"str1": str1, "str2": str2})
+        expected = json.dumps([
+            json.dumps({"key": "str1", "objType": "SEGSTRING", "dtype": "str", "val": str1.name}),
+            json.dumps({"key": "str2", "objType": "SEGSTRING", "dtype": "str", "val": str2.name})
+        ])
+        assert args == expected
+
+        size, args = _json_args_to_str({"str_list": [str1, str2]})
+        expected = json.dumps([
+            json.dumps({"key": "str_list", "objType": "LIST", "dtype": ak.Strings.objType,
+                        "val": json.dumps([str1.name, str2.name])}),
+        ])
+        assert args == expected
+
+    def test_dict_arg(self):
+        data = {
+            "json_1": {
+                "param1": 1,
+                "param2": "abc",
+                "param3": [1, 2, 3],
+                "param4": ak.arange(10),
+                "param5": ak.array(["abc", "123"])
+            }
+        }
+        size, args = _json_args_to_str(data)
+        expected = json.dumps([
+            json.dumps({"key": "json_1", "objType": "DICT", "dtype": "dict", "val": json.dumps([
+                json.dumps({"key": "param1", "objType": "VALUE", "dtype": ak.resolve_scalar_dtype(data["json_1"]["param1"]), "val": "1"}),
+                json.dumps({"key": "param2", "objType": "VALUE", "dtype": ak.resolve_scalar_dtype(data["json_1"]["param2"]), "val": "abc"}),
+                json.dumps({"key": "param3", "objType": "LIST", "dtype": ak.resolve_scalar_dtype(data["json_1"]["param3"][0]), "val": json.dumps([str(x) for x in data["json_1"]["param3"]])}),
+                json.dumps({"key": "param4", "objType": "PDARRAY", "dtype": str(data["json_1"]["param4"].dtype), "val": data["json_1"]["param4"].name}),
+                json.dumps({"key": "param5", "objType": "SEGSTRING", "dtype": "str", "val": data["json_1"]["param5"].name}),
```

Comment:
we use `data["json_1"]` a lot in this chunk. I'm wondering if it might be cleaner to save it off? I don't have super strong feelings about it tho

Reply:
I thought about that as well, but wasn't sure. I went ahead and updated it.